getting-started/optional_installation_settings.md

```diff
@@ -0,0 +1,220 @@
+
+        # Optional: The paths for extensions may be in different places, please refer to each extension
+        # and add the factories you require without the ".rb" extension, avoid the directory
+        # as it's supported by FactoryBot but can generate ambigous paths when a file with the
+        # same name exists.
+        extension_paths = []
+        # Optional:
+        # [ alt_1, alt_2, alt_3].map { |path| path.chomp(".rb") }
+
+        # Where:
+        # alt_1: MySolidusExtension::Engine.root.join("lib/solidus_content/factories.rb),
+        # alt_2: MySolidusExtension::Engine.root.join("lib/solidus_content/factories/product.rb"),
+        # alt_3: MySolidusExtension::Engine.root.join("lib/solidus_content/factories/product_factory.rb"),
+        # etc.
+
+        # The application factories, according to the place in which they're stored.
+        app_paths = [
+          Rails.root.join('lib', 'factories'),
+          Rails.root.join('spec', 'factories'),
+        ]
+
+        FactoryBot.definition_file_paths = solidus_paths + extension_paths + app_paths
+      end
+    end
+  end
+end
+```
+{% endcode %}
+
+Finally, modify the configuration to reference the customized methods which will give you the ability to create/build factories and stub authentication:
+
+{% code title="spec/spec\_helper.rb" %}
+```ruby
+require 'spree/testing_support/authorization_helpers'
+
+RSpec.configure do |config|
+  #...
+  config.include FactoryBot::Syntax::Methods
+end
+```
+{% endcode %}
+
+### Byebug
+
+[Byebug](https://github.com/deivid-rodriguez/byebug) is debugging tool which will help you investigate and understand code while it executes. There is no other installation instructions required other than to simply insert `byebug` method anywhere in the code base while you run a test or navigate through your development server. This will pause the execution of code where the method was inserted and provide you with a debugging interface in your shell. We highly recommend familiarizing yourself with byebug's [guide](https://github.com/deivid-rodriguez/byebug/blob/master/GUIDE.md) and [commands](https://github.com/deivid-rodriguez/byebug#byebugs-commands) to learn how to get the most out of this powerful tool.
+### Capybara and Webdrivers
+
+[Capybara](https://github.com/teamcapybara/capybara) helps you test web applications by simulating how a real user would interact with your app. This is complimented by the `webdrivers` gem to provide selenium drivers required by Capybara to test multiple browsers.  However, some setup is required to utilize the features we will be testing.
+
+First, let's setup our support file structure for RSpec. From the application's root folder, navigate to spec and create a new folder `support/helpers` and in that new folder, create a new file `capybara_config.rb` with the following contents:
+
+{% code title="spec/helpers/capybara\_config.rb" %}
```

Comment:
I'm reluctant to add installation instructions for third-party gems, as it adds a burden to us to keep them updated. Thoughts @kennyadsl ?

Reply:
I agree, if we can link to projects' specific instruction instead it would be better. Unless we have some special configuration to do for Solidus to work.

Reply:
I removed most of the installation instructions, but left the built code as an example of the setup I used to get it to work. Let me know if this is still too much and if I should remove it all together. 
Does this comment extend to DatabaseCleaner section as well?

Thanks!

Reply:
Yes, in my view, it applies to DatabaseCleaner, too. But you're right that if we're following a manual, then we need to add the code.